### PR TITLE
GenerateUniqueName for ReplaceMethodWithProperty

### PR DESCRIPTION
**Pull request by YairHalberstadt**
Fixes https://github.com/dotnet/roslyn/issues/42699

### CODE REVIEW

File: src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs [hunk 1]
```diff
@@ -2335,6 +2335,29 @@ int GetGoo()
 }");
         }
 
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty), WorkItem(42699, "https://github.com/dotnet/roslyn/issues/42699")]
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.cs]:
```suggestion
        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)]
        [WorkItem(42699, "https://github.com/dotnet/roslyn/issues/42699")]
```

File: src/EditorFeatures/VisualBasicTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.vb [hunk 2]
```diff
@@ -911,5 +911,22 @@ end class")
     End Function
 end class")
         End Function
+
+        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty), WorkItem(42699, "https://github.com/dotnet/roslyn/issues/42699")>
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.vb]:
```suggestion
        <Fact, Trait(Traits.Feature, Traits.Features.CodeActionsReplaceMethodWithProperty)>
        <WorkItem(42699, "https://github.com/dotnet/roslyn/issues/42699")>
```

File: src/EditorFeatures/VisualBasicTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.vb [hunk 3]
```diff
@@ -911,5 +911,22 @@ end class")
...
+        Public Async Function TestSameNameMemberAsProperty() As Task
+            Await TestInRegularAndScriptAsync(
+"class C
+    Public Goo as integer
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/VisualBasicTest/CodeActions/ReplaceMethodWithProperty/ReplaceMethodWithPropertyTests.vb]: can you add a VB case where the existing member is cased differently?  i.e. `goo`.

**Comment by CyrusNajmabadi** [comment]
LMK when this passes.  I'll merge in.  Thanks!

**Comment by YairHalberstadt** [comment]
Linux test coreClr was cancelled. Can you rerun it please @CyrusNajmabadi ?

**Comment by YairHalberstadt** [comment]
@CyrusNajmabadi Tests passed

**Pull request merged**